graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -258,11 +288,18 @@ public GraphqlService build() {
             configurer.configure(builder);
         }
 
-        final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
-        final List<Consumer<? super DataLoaderRegistry>> dataLoaderRegistries =
-                dataLoaderRegistryConsumers.build();
-        for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistries) {
-            configurer.accept(dataLoaderRegistry);
+        final Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory;
+        if (this.dataLoaderRegistryFactory != null) {
+            dataLoaderRegistryFactory = this.dataLoaderRegistryFactory;
+        } else if (dataLoaderRegistryConsumers != null) {
+            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+            for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistryConsumers.build()) {
+                configurer.accept(dataLoaderRegistry);
+            }
+            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
+        } else {
+            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
```

Comment:
As per https://github.com/graphql/dataloader#getting-started, `DataLoaderRegistry` is a request scope object, we should not reuse it.
> Typically instances are created per request when used within a web-server like [express](http://expressjs.com/)
```suggestion
            dataLoaderRegistryFactory = ctx -> new DataLoaderRegistry();
```

Reply:
Yes, it is.
However, to reduce confusion for existing users, the specification has been kept and new methods have been added.

Reply:
If the previous behavior was a bug, we probably should just fix it?

Reply:
I fixed it with the code by @ikhoon  suggested :)